pkg/apis/terraform/v1alpha1/provider_types.go

```diff
@@ -168,6 +172,37 @@ type ProviderSpec struct {
 	Summary string `json:"summary,omitempty"`
 }
 
+// JobLabels returns the labels which are automatically added to all jobs
+func (p *Provider) JobLabels() map[string]string {
+	if p.Spec.Job == nil {
+		return nil
+	}
+
+	return p.Spec.Job.Labels
+}
+
+// JobAnnotations returns the annotations which are automatically added to all jobs
+func (p *Provider) JobAnnotations() map[string]string {
+	if p.Spec.Job == nil {
+		return nil
+	}
+
+	return p.Spec.Job.Annotations
+}
+
+// ProviderJobSelectors is a collection of labels and annotations which are automatically
+// added to all jobs whom are created and use this provider. This can be useful to inject
+// cloud vendor specific labels and annotations to the jobs; Azure workload identity, or
+// AWS IAM roles for service accounts.
+type ProviderJobSelectors struct {
```

Comment:
nitpick: I wouldn't call this struct 'Selectors' as they don't select, they define

Reply:
changed to` JobMetadata`